src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -20,7 +20,9 @@ namespace System.Text.RegularExpressions
 {
     public abstract class RegexRunner
     {
-        protected internal int runtextbeg;         // beginning of text to search
+        protected internal int runtextbeg;         // beginning of text to search. We now always use a sliced span of the input
+                                                   // from runtextbeg to runtextend, which means that runtextbeg is now always 0 except
+                                                   // for CompiledToAssembly scenario which works over the original input.
         protected internal int runtextend;         // end of text to search
```

Comment:
You updated the comment for runtextbeg.  Is there something similar to say for runtextend, e.g. it's always equal to the length of the span input except in CompiledToAssembly?

Reply:
I intentionally didn't change this one since the value and the way it's used is still essentially the same. The comment doesn't specify if "text to search" is the whole input, and I kind of thought it more closely meant "the portion of the text that you actually want to search". runtextbeg on the other hand had a major change primarily on its value which is always 0 now, so I thought of adding a comment to it for future reference in case somebody else wanted to use it. All that said, I'm fine if you want me to be more verbose in the runtexend comment too.

Reply:
To my eyes, runtextbeg and runtextend are a pair, so the fact that runtextbeg has this big explanatory comment and runtextend doesn't makes me start wondering what's special or not special about runtextend.  At least a bit more info in a comment would be helpful, to me at least.

Reply:
Sounds good, I'll add a more verbose explanation.